docs/DeveloperGuide.md

```diff
@@ -983,38 +983,502 @@ testers are expected to do more *exploratory* testing.
+       Expected: All internship applications with `Offered` status are shown.
+    
+    3. Test case: `filter s`<br>
+        Expected: Internship list is not filtered. Error details shown in error message.
+
+2. Filtering internships by status while internships are being filtered.
+
+    1. Prerequisites: Filter the internship list using either `findc`, `findp` or `findt` command. Multiple internships in the list.
+
+    2. Test case: Similar to previous.<br>
+       Expected: Similar to previous. Filters do not stack on one another.
+
+### Sorting internships 
+
+Command: `sort` <br>
+More information on usage: [sort command](UserGuide.html#sorting-internship-applications-sort)
+
+1. Sorting internships while all internships are being shown.
+
+    1. Prerequisites: List all internships using the `list` command. Multiple internships in the list.
+
+    2. Test case: `sort time a`<br>
+       Expected: The currently displayed internships are sorted in ascending manner by upcoming tasks, with the 
+       internship with the task with the earliest date and time that is after the current date and time at the top.
+
+    3. Test case: `sort salary a`<br>
+       Expected: The currently displayed internships are sorted in ascending manner by salary, with the internship with the lowest 
+       salary at the top.
+   
+    4. Test case: `sort salary j`<br>
+       Expected: Internship list is not sorted. Error details shown in error message.
+
+2. Sorting internships while internships are being filtered.
+
+    1. Prerequisites: Filter the internship list using either `findc`, `findp` or `findt` command. Multiple internships in the list.
+
+    2. Test case: Similar to previous.<br>
+       Expected: Similar to previous. 
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+   1. Open `InTrack.jar` and make any changes to the internship list with the commands provided, being sure to leave at 
+   least one internship in the list.
+   
+   3. Edit the `data/intrack.json` file by making any one of the following changes before saving the file and reopening `InTrack.jar`.
+   
+      1. Test case: Edit the salary field of the first internship to `invalid`.
+      Expected: InTrack starts with an empty internship list.
+      
+      2. Test case: Edit the status field of the first internship to `invalid`.
+      Expected: Similar to previous.
+      
+      3. Test case: Edit the email field of the first internship to `invalid`.
+      Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+2. Dealing with missing files.
+   
+   1. Test case: Exit InTrack, then delete the `data/intrack.json` file. Reopen InTrack.
+   Expected: All internships are deleted. InTrack will start as expected with sample data provided.
+   
+   2. Test case: Exit InTrack, then delete the `config.json` file. Reopen InTrack.
+   Expected: InTrack starts as expected.
```

Comment:
Maybe it would be better to clarify a bit more on what you mean by "InTrack starts as expected". What would be the expected behaviour?